### PR TITLE
Fix settings not updating on reload

### LINKED ISSUE #9280: Appearance settings in a pane aren't updating on reload
I had a background image in a profile and then tried to update it in the settings UI while the profile was open in a pane, and the image didn't update. Same thing happened when I tried to update my color scheme while the profile was open in a pane. (I had two panes open at the time.)

Looks like this also happens when you try to enable `focusFollowMouse` using the settings UI. The setting is just ignored.

# Environment

```none
Windows build number: 10.0.21318.0
Windows Terminal version (if applicable): 1.7.543.0

Any other software?
```

# Steps to reproduce

Open two panes and change the background image or color scheme in the settings UI.

# Expected behavior

The pane updates its appearance to the new settings.

# Actual behavior

The newly saved settings are being ignored and the pane looks the same.

### PR DESCRIPTION
**Pull request by PankajBhojwani**
## Summary of the Pull Request
Fix for #9280 

In #8602, we started passing a child of the `TerminalSettings` to the control upon tab initialization, but forgot to do the same when new controls get created on a pane split. 

## PR Checklist
* [x] Closes #9280 
* [x] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/Terminal) and sign the CLA
* [ ] Tests added/passed
* [ ] Documentation updated. If checked, please file a pull request on [our docs repo](https://github.com/MicrosoftDocs/terminal) and link it here: #xxx
* [ ] Schema updated.
* [x] I work here

## Validation Steps Performed
Settings reload with multiple panes works

### CODE REVIEW

File: src/cascadia/TerminalApp/TerminalPage.cpp [hunk 1]
```diff
@@ -876,7 +876,7 @@ namespace winrt::TerminalApp::implementation
 
         if (debugConnection) // this will only be set if global debugging is on and tap is active
         {
-            TermControl newControl{ settings, debugConnection };
+            TermControl newControl{ *(winrt::get_self<TerminalSettings>(settings)->CreateChild()), debugConnection };
```
Comment (PankajBhojwani) [inline src/cascadia/TerminalApp/TerminalPage.cpp:879]: This change doesn't affect the settings reload bug but I figured we should update this call as well

File: src/cascadia/TerminalApp/TerminalPage.cpp [hunk 2]
```diff
@@ -1840,7 +1840,7 @@ namespace winrt::TerminalApp::implementation
                 return;
             }
 
-            TermControl newControl{ controlSettings, controlConnection };
+            TermControl newControl{ *(winrt::get_self<TerminalSettings>(controlSettings)->CreateChild()), controlConnection };
```
Comment (DHowett) [inline src/cascadia/TerminalApp/TerminalPage.cpp:1843]: Oh gosh. Can you file a Code Health task to clean up all the different places in which we create a TerminalControl with its settings? I don't love that we had to make this fix in multiple places.

File: src/cascadia/TerminalApp/TerminalPage.cpp (hunk 1 above)
Comment (DHowett) [reply src/cascadia/TerminalApp/TerminalPage.cpp:879]: It actually will impact reload, if the pane is a debug tap pane :D

**Review by DHowett — APPROVED** [review]

**Comment by carlos-zamora** [comment]
I think you meant to close #9280?

**Comment by PankajBhojwani** [comment]
> I think you meant to close #9280?

Yes... thank you for the catch

**Review by carlos-zamora — APPROVED** [review]

**Review by zadjii-msft — APPROVED** [review]

**Pull request merged**